enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/discovery/HazelcastClient.java

```diff
@@ -37,6 +39,9 @@
 import static org.neo4j.causalclustering.discovery.HazelcastClusterTopology.READ_REPLICA_BOLT_ADDRESS_MAP_NAME;
 import static org.neo4j.causalclustering.discovery.HazelcastClusterTopology.READ_REPLICA_MEMBER_ID_MAP_NAME;
 import static org.neo4j.causalclustering.discovery.HazelcastClusterTopology.READ_REPLICA_TRANSACTION_SERVER_ADDRESS_MAP_NAME;
+import static org.neo4j.causalclustering.discovery.HazelcastClusterTopology.SERVER_TAGS_MULTIMAP_NAME;
```

Comment:
I think tagging could be seen as a general concept for identifying servers or sets of servers, and you could imagine building several things on top of that, e.g. load balancing and replication policies, some future security policy, general affinity, etc. So from that perspective I don't think the layering is bad.

If you look at the setting I simply named it `causal_clustering.server_tags` which is inline with this way of looking at it (e.g. no load_balancing in the name).

Is tagging really the most appropriate design for identification meant for other purposes? Future will have to show.